cmake/OpenMP.cmake

```diff
@@ -23,7 +23,7 @@ endif()
 set(OpenMP_cmake_included true)
 include("cmake/Threading.cmake")
 
-if (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+if (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
```

Comment:
Currently this check applied to all Clangs on macOS, after the change only to Apple provided Clang. If we want to retain the old logic, then `MATCHES` would be a better option like: `if (APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")`. It should catch both Clangs.